File: ALGORITHM/programmers/22년9월/N개의_최소공배수.js
function solution(arr) {
  // 최대공약수
  const gcd = (a, b) => {
      if (a > b) {
          const tmp1 = a
          a = b
          b = tmp1
      }
      while (a > 0) {
          const tmp2 = b
          b = a
          a = tmp2 % b
      }
      return b
  }
  // 최소공배수
  let lcm = arr[0]
  arr.map(n => lcm = lcm*n/gcd(lcm,n))
  return lcm
}

File: ALGORITHM/programmers/22년9월/N개의_최소공배수.py
def lcm(x, y):
    c = x * y
    a, b = max(x, y), min(x, y)
    while a > 0:
        a, b = b%a, a
    return c/b

def solution(arr):
    LCM = arr[0]
    for n in arr:
        LCM = lcm(LCM, n)
    return LCM